src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -261,12 +262,19 @@ private void FireOnStarting()
             {
                 foreach (var entry in onStarting)
                 {
-                    entry.Key.Invoke(entry.Value).Wait();
+                    try
```

Comment:
This try/catch should be outside the foreach, we don't want to keep executing the OnStarted callbacks if the request has failed. OnCompleted is different.